starknet_devnet/blocks.py

```diff
@@ -103,8 +112,8 @@ def __assert_block_number_in_range(self, block_number: BlockIdentifier):
                 code=StarkErrorCode.MALFORMED_REQUEST, message=message
             )
 
-        if block_number >= self.get_number_of_blocks():
-            message = f"Block number too high. There are currently {len(self.__num2block)} blocks; got: {block_number}."
+        if block_number >= self.get_number_of_not_aborted_blocks():
```

Comment:
I don't like it that we are executing get_number_of_not_aborted_blocks multiple times here.

Reply:
true, fixed in cf60a08603da192c7030c4479e383b187f717a80